### DISCUSSION TITLE
Weird behavior when moving window on macOS (and crash?)

### DISCUSSION THREAD

**Discussion by OfficialCRUGG** [ppy/osu#28069] [Q&A] [answered]
### Type

Other

### Bug description

When trying to move the game window on macOS it behaves very oddly. See the attached video for details.
I also managed to have osu! crash on me whilst moving the window, but I don't know how to reproduce this.
If you need additional details to locate the source of this issue, I'm happy to provide whatever is needed

This is a very far stretch, but I feel like this _could_ possibly be related to my other issue #28053?

### Screenshots or videos

https://github.com/ppy/osu/assets/25248999/56d8a9d1-afd3-45d7-a209-8634ca0f14de

### Version

2024.412.1-lazer

### Logs

[compressed-logs.zip](https://github.com/ppy/osu/files/15180106/compressed-logs.zip)

**Comment by bdach** [comment]
Logs look clean except one that got truncated, so probably a driver/OS-level crash. Please check if there is anything in Console that may look relevant.

**Comment by OfficialCRUGG** [comment]
I'm not quite sure where to find that?

Also, I've done some further testing and I'm very very sure that my initial assumption of this being related to #28053 (cursor snapping to center when moving inside game window) was correct.
If I move the window very slowly, nothing happens. Only if I move the window faster, this occurs. And because the window movement on macOS is always lagging a bit behind the cursor, I assume my cursor shortly entered the actual game window, causing the cursor centering described in that other issue, and thus moving the window there.

**Comment by bdach** [comment]
> I'm not quite sure where to find that?

https://support.apple.com/guide/console/welcome/mac

Either way this looks like something going wrong inside sdl.

**Comment by peppy** [comment, marked as answer]
probably same as other issue opened by same user – setting mouse mode to unsupported mode (unsupported due to issues, clear message in settings saying this)

![2024-05-02 22 57 54@2x](https://github.com/ppy/osu/assets/191335/c44c981d-d06c-4405-b183-43aa015f982c)

**Reply by OfficialCRUGG** [reply]
Yeah, it actually is that. (not sure why I didn't think of directly trying when I made the assumption the two could be related)
Moving the window too quickly causes the cursor to go inside the game window a tiny bit, which causes the centering of the cursor with high precision mouse turned on, and since the window is still in the "being dragged" state, it gets placed there.

I actually didn't remember having that feature turned on, but I believe I turned it on because someone told me this would counter macOS's mouse acceleration.
